src/audio/module_adapter/module_adapter.c

```diff
@@ -1223,6 +1223,17 @@ void module_adapter_free(struct comp_dev *dev)
 		buffer_free(buffer);
 	}
 
+	list_for_item_safe(listp, _listp, &mod->priv.memory.mem_list) {
+		struct module_memory *mem = container_of(listp, struct module_memory, mem_list);
+
+		rfree(mem->ptr);
+		list_item_del(&mem->mem_list);
+		rfree(mem);
+		err_count++;
```

Comment:
is this a bug or is this a feature? I thought that the purpose of this managed allocation was the same as in the Linux kernel - you allocate what you need per "module" and in the end it's all freed automatically when the module is freed?

Reply:
Warning removed, and redundant - init failure branch or src_free() calls to mod_free() - removed.